influxdb3_catalog/src/catalog.rs

```diff
@@ -862,12 +863,18 @@ impl TableDefinition {
             );
             column_map.insert(*col_id, name.into());
         }
-        schema_builder.with_series_key(series_key.clone().into_iter().map(|id| {
-            column_map
-                .get_by_left(&id)
-                // NOTE: should this be an error instead of panic?
-                .expect("invalid column id in series key definition")
-        }));
+        let series_key_names = series_key
```

Comment:
Can this be built within the `for` loop above? Looks like there's already some checks going on whilst building the column/columns map.

Reply:
It might be, but it is easier to construct the `Vec<Arc<str>>` of series key column names after the whole column `name`<>`id` map is constructed. The size of the series key isn't probably going to be very many in most cases and this only happens on table creation, so not often. I think it is okay to have separate iterations in this case.

Reply:
> this only happens on table creation

ah ok - since you mentioned it's in hot path, I assumed this is getting called for every write. I get it now, without this refactor the series key lookup happens every time in the write path.

Reply:
Sorry, that may have been a bit confusing. The `TableDefinition::series_key` method (which this PR splits into `TableDefinition::series_key_ids` and `TableDefinition::series_key_names`) _used_ to be in the hot path in the v3 write API (which was the reason for https://github.com/influxdata/influxdb/issues/25606). This PR removes the v3 write API, as part of https://github.com/influxdata/influxdb/issues/25611, so the method call actually isn't even used in OSS anymore (it is used in pro by the compactor, though I don't think that it is in a hot path). I figured while I was in there I might as well address https://github.com/influxdata/influxdb/issues/25606.